docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,119 @@
+This encapsulates slightly more advanced filter logic over the existing clusterdeployment labels. In the example above, hive_foo_counter metric will only be reported for the cluster deployments that are labelled with aro-snowflake, not in limited support, and has the metric value > 10m.
+
+### Implementation Details / Notes
+- In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.`
+- Current configuration of using a shorthand camelCase key to refer to a corresponding metric while configuring hiveconfig.spec.metricsConfig.metricsWithDuration will be removed, and users will have to use the full metric name in the proposed configuration.
+- Additionally, we would also want to remove `cluster_type` as a fixed label for all the metrics that do report it now. This would be a breaking change as the consumers who want `cluster_type` will have to configure it via metricsConfig.clusterDeploymentRelatedMetrics[].additionalClusterDeploymentLabels for all the metrics they need it for.
+- There are other fixed labels (like platform and version), which are directed fed from the clusterdeployment labels. Ideally we would want to feed them via hiveconfig, but we might need an exception for the cluster deployment labels that are applied by hive. 
+- All the options configured for a metric within clusterDeploymentRelatedMetrics, will work in tandem with each other. For ex, if all possible options are specified for a metric, then that metric will be reported with the additional labels as per additionalClusterDeploymentLabels, and will be reported only if it matches the labels and/or expressions as per clusterDeploymentLabelSelector and if the duration to be reported exceeds the minimumDuration
+- Failure modes:
+  These situations will cause a panic().
+  - clusterDeploymentRelatedMetrics[].name doesn't exist as a metric
+  - clusterDeploymentRelatedMetrics[].name is a metric without duration but minimumDuration was specified
+  - additionalClusterDeploymentLabels conflict with existing fixed label for that metric
+
+### Risks and Mitigations
+The biggest risk is the sheer number of metrics that are going to be affected.
+
+Many metrics related to cluster deployment which do not currently have support for additionalClusterDeploymentLabels will now have that support. This might create an avenue for worsening the cardinality, however, that would be the responsibility of Hive admin.
+
+It can get confusing for consumers to know the fixed labels each metric has.Update the [hive_metrics](https://github.com/openshift/hive/blob/master/docs/hive_metrics.md) doc to list out all labels and keep it up-to-date.
```

Comment:
...or have zero fixed labels. The obvious disadvantage being that the user has to configure a given label on each and every one if they want it across the board.

I anticipate SD reflexively asserting that they'll want labels like platform and version for every metric. But I think if we challenge that assertion, they're pretty likely to walk it back.

Reply:
Also - I hope uses see the hassle of mentioning multiple metrics in hiveConfig as enough of a deterrent to not apply additionalLabels to all metrics.

Reply:
Ahh, okay, the clarifying point is that the optional labels are always coming from ClusterDeployment.Labels.

What about if we said we will never have a fixed label that corresponds to a CD label? Does that work?

Reply:
[Later] I see now where you've said exactly this, cool ✓